File: Week-03/Day-2/EncodedLines.ts
'use strict';

// Create a method that decrypts encoded-lines.txt

let fs = require('fs');

function reversedLines(fileName: string) {
  let content: string[] = fs.readFileSync(fileName, 'utf-8').split('\r');
  let array: string[][] = [];
  let sentences: string = '';
  content.forEach(function (value) {
    array.push(value.split(''))
  })
  for (let x: number = 0; x < array.length; x++) {
    for (let y: number = 0; y < array[x].length; y++) {
      if (array[x][y] === '\n' || array[x][y] === ' ') {
        sentences += array[x][y]
      } else {
        sentences += String.fromCharCode(array[x][y].charCodeAt(0) - 1)
      }
    }
  }
  console.log(sentences);
}
reversedLines('EncodedLines.txt')

File: Java/Week3/Day-1/EncodedLines.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.HashMap;

public class EncodedLines {

  public static void main(String[] args) {
    // Create a method that decrypts encoded-lines.txt
    Path filePath1 = Paths.get("encoded-lines.txt");
    System.out.println(decode(filePath1));
  }

  public static String decode(Path pathInput) {
    String decrypted = "";
    try {
      char[] content = Files.readAllLines(pathInput).toString().toCharArray();
      for (int i = 0; i < content.length - 1; i++) {
        if ((int) content[i] == 32) {
          decrypted += " ";
        } else {
          decrypted += (char) ((int) (content[i] - 1));
        }
      }
    } catch (IOException e) {
      System.out.println("Can't find file.");
    } catch (Exception e) {
      System.out.println("Error");
    }
    decrypted = decrypted.substring(1);
    decrypted = decrypted.replace("+", "\n");
    return decrypted;
  }
}